dotnet/src/Connectors/Connectors.AI.PaLM/TextEmbedding/PaLMTextEmbeddingGeneration.cs

```diff
@@ -0,0 +1,182 @@
+    /// <param name="endpoint">Endpoint for service API call. If not specified, the base address of the HTTP client is used.</param>
+    public PaLMTextEmbeddingGeneration(string model, HttpClient httpClient, string? apiKey, string? endpoint = null)
+    {
+        Verify.NotNullOrWhiteSpace(model);
+        Verify.NotNull(httpClient);
+
+        this._model = model;
+        this._endpoint = endpoint;
+        this._httpClient = httpClient;
+        this._apiKey = apiKey;
+        if (httpClient.BaseAddress == null && string.IsNullOrEmpty(endpoint))
+        {
+            throw new AIException(
+                AIException.ErrorCodes.InvalidConfiguration,
+                "The HttpClient BaseAddress and endpoint are both null or empty. Please ensure at least one is provided.");
+        }
+
+        this._disposeHttpClient = false; // We should not dispose custom HTTP clients.
+    }
+
+    /// <inheritdoc/>
+    public async Task<IList<Embedding<float>>> GenerateEmbeddingsAsync(IList<string> data, CancellationToken cancellationToken = default)
+    {
+        return await this.ExecuteEmbeddingRequestAsync(data, cancellationToken).ConfigureAwait(false);
+    }
+
+    
+
+    #region private ================================================================================
+
+    /// <summary>
+    /// Performs HTTP request to given endpoint for embedding generation.
+    /// </summary>
+    /// <param name="data">Data to embed.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>List of generated embeddings.</returns>
+    /// <exception cref="AIException">Exception when backend didn't respond with generated embeddings.</exception>
+    private async Task<IList<Embedding<float>>> ExecuteEmbeddingRequestAsync(IList<string> data, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var embeddingRequest = new TextEmbeddingRequest
+            {
+                Text = string.Join(" ", data)
+            };
+
+            using var httpRequestMessage = new HttpRequestMessage()
+            {
+                Method = HttpMethod.Post,
+                RequestUri = this.GetRequestUri(),
+                Content = new StringContent(JsonSerializer.Serialize(embeddingRequest)),
+            };
+
+            httpRequestMessage.Headers.Add("User-Agent", HttpUserAgent);
+
+            var response = await this._httpClient.SendAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
+            var body = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+
+            var embeddingResponse = JsonSerializer.Deserialize<TextEmbeddingResponse>(body);
+
+            return new List<Embedding<float>>() { new Embedding<float>(embeddingResponse?.embedding.value) };
+        }
+        catch (Exception e) when (e is not AIException && !e.IsCriticalException())
```

Comment:
This "e is not AIException" additional check is not necessary since it's covered by the "e.IsCriticalException()" one.

Reply:
fixed